stop_and_wait_plugin/src/stop_and_wait_plugin.cpp

```diff
@@ -373,26 +358,25 @@ std::vector<cav_msgs::TrajectoryPlanPoint> StopandWait::compose_trajectory_from_
   {
     if (times[i] != 0 && !std::isnormal(times[i]) && i != 0)
     {  // If the time
-      ROS_WARN_STREAM("Detected non-normal (nan, inf, etc.) time. Making it same as before: " << times[i-1]);
+      RCLCPP_INFO_STREAM(nh_->get_logger(),"Detected non-normal (nan, inf, etc.) time. Making it same as before: " << times[i-1]);
       // NOTE: overriding the timestamps in assumption that pure_pursuit_wrapper will detect it as stopping case
       times[i] = times[i - 1];
     }
   }
 
   std::vector<double> yaws = carma_wm::geometry::compute_tangent_orientations(raw_points);
 
-
   for (size_t i = 0; i < points.size(); i++)
   {
-    ROS_DEBUG_STREAM("1d: " << downtracks[i] << " t: " << times[i] << " v: " << filtered_speeds[i]);
+    RCLCPP_INFO_STREAM(nh_->get_logger(),"1d: " << downtracks[i] << " t: " << times[i] << " v: " << filtered_speeds[i]);
   }
 
   auto traj = trajectory_from_points_times_orientations(raw_points, times, yaws, start_time);
 
-  while (traj.back().target_time - traj.front().target_time < ros::Duration(config_.minimal_trajectory_duration))
+  while (rclcpp::Time(traj.back().target_time) - rclcpp::Time(traj.front().target_time) < rclcpp::Duration(config_.minimal_trajectory_duration))
   {
-    cav_msgs::TrajectoryPlanPoint new_point = traj.back();
-    new_point.target_time = new_point.target_time + ros::Duration(config_.stop_timestep);
+    carma_planning_msgs::msg::TrajectoryPlanPoint new_point = traj.back();
+    new_point.target_time = rclcpp::Time(new_point.target_time) + rclcpp::Duration(config_.stop_timestep);
```

Comment:
`config_.stop_timestep` is `double` so rclcpp::Duration(double) is in nanoseconds. So it needs to be config_.stop_timestep * 1e9`

Good rule of thumb to check these kind of error is to search your packages folder using your VSCode search every instances of `Duration(` or `Time(` (pay attention to the parenthesis). OR also instances of `rclcpp::Duration ` or `rclcpp::Time ` (also pay attention to the space). And once you found it, edit with `1e9` or other constructors as you see fit.